src/EnergyPlus/DataZoneEquipment.cc

```diff
@@ -1307,20 +1299,23 @@ Real64 EquipList::SequentialCoolingFraction(EnergyPlusData &state, const int equ
 
 int GetZoneEquipControlledZoneNum(EnergyPlusData &state, DataZoneEquipment::ZoneEquip const ZoneEquipTypeNum, std::string const &EquipmentName)
 {
+    static constexpr std::string_view RoutineName("GetZoneEquipControlledZoneNum: ");
     int ControlZoneNum = 0;
 
     for (int CtrlZone = 1; CtrlZone <= state.dataGlobal->NumOfZones; ++CtrlZone) {
         if (!state.dataZoneEquip->ZoneEquipConfig(CtrlZone).IsControlled) continue;
         for (int Num = 1; Num <= state.dataZoneEquip->ZoneEquipList(CtrlZone).NumOfEquipTypes; ++Num) {
-            if (UtilityRoutines::SameString(EquipmentName, state.dataZoneEquip->ZoneEquipList(CtrlZone).EquipName(Num)) &&
-                ZoneEquipTypeNum == state.dataZoneEquip->ZoneEquipList(CtrlZone).EquipTypeEnum(Num)) {
-                ControlZoneNum = CtrlZone;
-                break;
+            if (ZoneEquipTypeNum == state.dataZoneEquip->ZoneEquipList(CtrlZone).EquipTypeEnum(Num) &&
+                UtilityRoutines::SameString(EquipmentName, state.dataZoneEquip->ZoneEquipList(CtrlZone).EquipName(Num))) {
+                return ControlZoneNum = CtrlZone;
             }
         }
-        if (ControlZoneNum > 0) break;
     }
-
+    ShowSevereError(state,
+                    fmt::format("{}{}=\"{}\" is not on any ZoneHVAC:Equipmentlist. It will not be simulated.",
```

Comment:
Error message moved here instead of scattered about in baseboard init functions. Other HVAC components can probably use similar changes.